phoenix-core/src/main/java/org/apache/phoenix/compile/ExpressionCompiler.java

```diff
@@ -200,12 +224,12 @@ public Expression visitLeave(ComparisonParseNode node, List<Expression> children
         Expression lhsExpr = children.get(0);
         Expression rhsExpr = children.get(1);
         PDataType dataTypeOfLHSExpr = lhsExpr.getDataType();
-        if (dataTypeOfLHSExpr != null && !dataTypeOfLHSExpr.isEqualitySupported()) {
+        if (!(lhsExpr instanceof BaseJSONExpression)&& dataTypeOfLHSExpr != null && !dataTypeOfLHSExpr.isEqualitySupported()) {
```

Comment:
The visitLeave(ComparisonParseNode) method gets called when =, <, >, etc. comparison operators are used in an expression. I thought this wasn't valid for JSON?